src/main/kotlin/com/terraformation/backend/seedbank/db/MigrateWithdrawnTotals.kt

```diff
@@ -0,0 +1,66 @@
+package com.terraformation.backend.seedbank.db
+
+import com.terraformation.backend.config.TerrawareServerConfig
+import com.terraformation.backend.customer.model.SystemUser
+import com.terraformation.backend.db.seedbank.tables.references.ACCESSIONS
+import com.terraformation.backend.db.seedbank.tables.references.WITHDRAWALS
+import com.terraformation.backend.log.perClassLogger
+import javax.inject.Named
+import org.jooq.DSLContext
+import org.jooq.impl.DSL
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.context.event.ApplicationStartedEvent
+import org.springframework.context.event.EventListener
+
+/**
+ * Migration to calculate total withdrawal weights and counts for accessions that don't already have
+ * them. This can be removed after it has run.
+ */
+@ConditionalOnProperty(TerrawareServerConfig.DAILY_TASKS_ENABLED_PROPERTY, matchIfMissing = true)
+@Named
+class MigrateWithdrawnTotals(
+    private val accessionStore: AccessionStore,
+    private val dslContext: DSLContext,
+    private val systemUser: SystemUser,
+) {
+  private val log = perClassLogger()
+
+  @EventListener
+  fun migrate(@Suppress("UNUSED_PARAMETER") event: ApplicationStartedEvent) {
+    val condition =
+        ACCESSIONS.TOTAL_WITHDRAWN_COUNT.isNull
+            .and(ACCESSIONS.TOTAL_WITHDRAWN_WEIGHT_GRAMS.isNull)
+            .andExists(
+                DSL.selectOne()
+                    .from(WITHDRAWALS)
+                    .where(WITHDRAWALS.ACCESSION_ID.eq(ACCESSIONS.ID))
+                    .and(WITHDRAWALS.WITHDRAWN_QUANTITY.isNotNull))
+
+    systemUser.run {
+      accessionStore.forEachAccession(condition) {
+        val accession = it.withCalculatedValues()
+
+        if (accession.totalWithdrawnCount != null || accession.totalWithdrawnWeight != null) {
```

Comment:
probably unlikely .. but in the case of a re-run do we want to skip those that already have it set?

Reply:
The search condition should do that, but I should make it clearer what's going on. Thanks.